cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java

```diff
@@ -111,15 +110,31 @@ public RuntimeMonitor(ProgramRunId programRunId, CConfiguration cConf, RuntimeMo
     this.programFinishTime = -1L;
     this.lastProgramStateMessages = new LinkedList<>();
     this.requestKeyToLocalTopic = createTopicConfigs(cConf);
+    this.metricScheduledService = metricScheduledService;
   }
 
   @Override
   protected ScheduledExecutorService executor() {
     return scheduledExecutorService;
   }
 
+  @Override
+  protected void doStartUp() {
+    try {
+      metricScheduledService.startAndWait();
+    } catch (Exception e) {
+      LOG.error("Failed to start metrics service for program run {}, node minute metrics will not be updated.",
```

Comment:
Should be just a warn since things would still keep going.

Reply:
fixed